service-worker.js

```diff
@@ -0,0 +1,19 @@
+---
+---
+
+importScripts('/scripts/sw-toolbox.js');
+
+var swOptions = {
+  debug: true
+}
+
+var preCachedRessources = [
+  '{{ "/css/main.css" | prepend: site.baseurl }}?t={{ site.time | date_to_xmlschema }}"',
+  {% for page in site.pages %}
+  '{{ page.url }}',
+  {% endfor %}
```

Comment:
I'm not sure if the following is valid. I see this to be valid at-least for page transitions from and to REPL.

The Service Worker shouldn't change from pageA to pageB if both the pages are under the scope of the same service worker. This is because, service worker updates based on the byte diff, and during a page change, a new service worker will install (if the resources are different -> i.e new `service-worker.js`).